new features: thread_rng, impl_rng_core, and versioned glam

gordlea opened the pull request:
* thread_rng adds support for thread local random number generation.
* impl_rng_core implements the RngCore and SeedableRng traits from rand.
* You can now choose which version of glam. you wish to use by using the glam_02x feature. 

This pr was created because I wanted to use frand in a bevy game I'm working on, but I needed the thread_rng support.

I was originally going to just fork the repo, but I thought I'd clean it up and open a PR to see what you thought.

I'm open to feedback or modifications, just thought you mind find this useful.

engusmaze commented:
Do you need multiple versions of glam, or just the latest ones?

engusmaze commented:
How about just making the glam version to 0, and then the version will be specified on the user side

gordlea commented:
I just wanted multi version compatibility because bevy often lags behind the latest by a few versions.

Re: version 0, that could work, I never thought of doing it that way. I'll give it a shot and update.

gordlea committed:
feat: new features: thread_rng, impl_rng_core

thread_rng adds support for thread local random number generation.

impl_rng_core implements the RngCore and SeedableRng traits from rand.

gordlea commented:
@engusmaze how does it look now?

gordlea committed:
fix: add missing cfg feature annotation

gordlea committed:
fix: add ability to get base rng from thread rng

engusmaze commented:
I've made a few changes. I like the idea of `ThreadRng`, it's much faster to have a per-thread rng than to recreate the rng every time you want to use it from somewhere else. So I made it so that `ThreadRand` doesn't depend on `rand_core`, but only depends on `std` because it needs to use `thread_local` and `Rc`. I also took `impl_rng_core` but split it into a different file.
I also want to ask, which part of `rand` did you need that you decided to make an implementation of `RngCore`?

gordlea commented:
I  wanted to be able to use (https://docs.rs/rand/latest/rand/distributions/) if i remember correctly. If there's an alternate way to do so I'd be happy to try to simplify things. 😄
